deps/libtracyclient.mk

```diff
@@ -0,0 +1,72 @@
+## LIBTRACYCLIENT ##
+ifneq ($(USE_BINARYBUILDER_LIBTRACYCLIENT),1)
+LIBTRACYCLIENT_GIT_URL:=https://github.com/wolfpld/tracy.git
+LIBTRACYCLIENT_TAR_URL=https://api.github.com/repos/wolfpld/tracy/tarball/$1
+$(eval $(call git-external,libtracyclient,LIBTRACYCLIENT,,,$(SRCCACHE)))
+
+LIBTRACYCLIENT_BUILDDIR := $(BUILDDIR)/$(LIBTRACYCLIENT_SRC_DIR)
+LIBTRACYCLIENT_SRCCACHE := $(SRCCACHE)/$(LIBTRACYCLIENT_SRC_DIR)
+
+LIBTRACYCLIENT_CMAKE :=
+LIBTRACYCLIENT_CMAKE += -DBUILD_SHARED_LIBS=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_FIBERS=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_NO_BROADCAST=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_ONLY_LOCALHOST=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_NO_CODE_TRANSFER=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_NO_FRAME_IMAGE=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_NO_CRASH_HANDLER=ON
+LIBTRACYCLIENT_CMAKE += -DTRACY_ON_DEMAND=ON
+
+$(LIBTRACYCLIENT_SRCCACHE)/cmake-patch-applied: $(LIBTRACYCLIENT_SRCCACHE)/source-extracted
+ifneq ($(OS),WINNT)
+	echo "target_compile_definitions(TracyClient PUBLIC __STDC_FORMAT_MACROS)" >> $(LIBTRACYCLIENT_SRCCACHE)/CMakeLists.txt
+else
+	echo "target_compile_definitions(TracyClient PUBLIC WINVER=0x0602 _WIN32_WINNT=0x0602)" >> $(LIBTRACYCLIENT_SRCCACHE)/CMakeLists.txt
+endif
+	echo 1 > $@
+
+$(LIBTRACYCLIENT_SRCCACHE)/libTracyClient-freebsd-elfw.patch-applied: $(LIBTRACYCLIENT_SRCCACHE)/cmake-patch-applied
+	cd $(LIBTRACYCLIENT_SRCCACHE) && \
+		patch -p1 -f < $(SRCDIR)/patches/libTracyClient-freebsd-elfw.patch
+	echo 1 > $@
+
+$(LIBTRACYCLIENT_SRCCACHE)/libTracyClient-no-crash-handler.patch-applied: $(LIBTRACYCLIENT_SRCCACHE)/libTracyClient-freebsd-elfw.patch-applied
+	cd $(LIBTRACYCLIENT_SRCCACHE) && \
+		patch -p1 -f < $(SRCDIR)/patches/libTracyClient-no-crash-handler.patch
+	echo 1 > $@
```

Comment:
@vtjnash this is a good example of a case where I think it makes more sense to unpack source within `$(BUILDDIR)`, as opposed to `$(SRCCACHE)`, since we're patching things.  Do you agree?

Reply:
Sounds good to me - I only did it this way because that's how we handle patches for LLVM right now.

I assume you effectively would like to see https://github.com/JuliaLang/julia/pull/48830 done for this Makefile?

Reply:
Yes, correct

Reply:
xref: https://github.com/JuliaLang/julia/issues/31530